### PR TITLE
Purchases: Move methods from `PurchasesMixin` to new, non-mixin module

### LINKED ISSUE #12109: Media Section: Selected images in media section remain selected in media modal in editor.
Steps to reproduce:

1. Go to `/media`.
2. Select some images.
3. Click "add" to create a post. (You can navigate to anywhere before you click "add" button in Blog Posts.)
4. Open the media modal.
5. You see the same images are already selected.

I don't think this is a feature... The context is different in the section and the modal, and it can be confusing that some images are already selected when you create a new post if you've forgotten that you've selected some images in the section a while ago.

untitled2

### PR DESCRIPTION
**Pull request by drewblaisdell**
This change affects the following components:

`CancelPrivateRegistration`
`CancelPurchase`
`ConfirmCancelPurchase`
`ManagePurchase`
`EditCardDetails`
`EditPaymentMethodCreditCard`
`EditPaymentMethod`
`EditPaymentMethodPaypal`

I was originally going to include this as part of #475, but it makes more sense as its own PR as it requires much more testing. I added this to the Purchases v1 milestone because we want to use some of these changes in that PR.

Fixes #12109.

**Testing**
- Assert that the visiting the following routes is possible for applicable sites/purchases/payment methods without errors:
  - `/purchases/:site/:purchase_id`
  - `/purchases/:site/:purchase_id/payment/edit/:payment_id` for payments by credit card/PayPal
  - `/purchases/:site/:purchase_id/cancel`
  - `/purchases/:site/:purchase_id/confirm-cancel`
  - `/purchases/:site/:purchase_id/cancel-private-registration`
- Assert that each aforementioned route is a valid entry point for Calypso by visiting the page, refreshing, and asserting that no error appeared.
- Assert that the 'back' link in each page's header works without errors.
- Assert that the link from a domain to the plan it is included in works without errors.

- [x] QA review
- [x] Code review

cc @fabianapsimoes @scruffian @stephanethomas

### CODE REVIEW

**Comment by gziolo** [comment]
This looks great @drewblaisdell. Excellent PR :+1:

File: client/me/purchases/cancel-private-registration/index.jsx [hunk 1]
```diff
@@ -132,7 +127,7 @@ const CancelPrivateRegistration = React.createClass( {
 		return (
```
Comment (gziolo) [inline client/me/purchases/cancel-private-registration/index.jsx]: On line 42 there is another `isDataLoading` method which also can be refactored.

**Comment by gziolo** [comment]
It works perfectly fine for me, I didn't test only `Cancel Private Registration` page. Please double check before merging.

**Comment by gziolo** [comment]
Still looking good :+1:

**Comment by scruffian** [comment]
LGTM :+1: 

I added a couple of commits. One to remove an extra `isDataLoading` and one to rename helper to utils, since we use this name more.

**Pull request merged**